lib/rouge/lexer.rb

```diff
@@ -504,12 +533,26 @@ def self.detect?(text)
   end
 
   module Lexers
-    @_loaded_lexers = {}
+    @_loaded = {}
+    def self.preload(tag)
+      lang = Lexer.find(tag)
+      if lang
+        lang.lexer_class
+      elsif !@_loaded.key?(tag)
+        @_loaded[tag] = true
+        # needed during generate:cache, and also common files
+        load "#{__dir__}/lexers/#{tag}.rb"
+        Lexer.last_registered
```

Comment:
Hm. Now that you mention it, there are a number of lexers that aren't in the file named by their tag. But we need to load the lexers in dependency order when building the cache to avoid double-loading. I'll have to think about this a bit.

Reply:
This has been resolved by the more recent changes, right?

Reply:
Yep! We don't even include code for the generator here anymore - it monkey-patches this method before running, so it can manually load the lexers in order.

Reply:
In fact, I've removed the `last_registered` method as it's no longer used.